src/main/resources/templates/vpc/private-subnet-resources.json.vtp

```diff
@@ -200,8 +200,41 @@
 			],
 			"SecurityGroupIds": [{ "Ref": "VPCEndpointSecurityGroup" }]
 		}
-	}
+	},
+    #foreach( $vpcEndpointSvc in $vpcEndpointServices )
+    "${vpcEndpointSvc.replaceAll("-", "")}VPCEndpoint": {
+        "Type":"AWS::EC2::VPCEndpoint",
+        "Properties": {
+           "PolicyDocument":{
+              "Version":"2012-10-17",
+              "Statement":[
+                 {
+                    "Effect":"Allow",
+                    "Principal": "*",
+                    "Action": "*",
+                    "Resource": "*"
+                 }
+              ]
+           },
+           "PrivateDnsEnabled": true,
+           "ServiceName": { "Fn::Sub": "com.amazonaws.#[[${AWS::Region}]]#.${vpcEndpointSvc}" },
+           "SubnetIds": [
+              #foreach( $privateSubnet in $subnetGroup.subnets )
+                #if ($vpcEndpointsAz.contains(${privateSubnet.availabilityZone}))
+                {"Ref": "${privateSubnet.name}"}
+                #if( $foreach.hasNext),#end
+                #end
+              #end
+           ],
+           "SecurityGroupIds": [{ "Ref": "VPCEndpointSecurityGroup" }],
+           "Tags": [ {"Key": "Name", "Value": "${vpcEndpointSvc}-vpce" } ],
```

Comment:
Why the tag?

Reply:
To give it a name in the console. I would have expected a Name property but it works in a roundabout way...

Reply:
Oh that's weird 😕